### PR TITLE
Consolidate Microsoft.NETCore.App property usage

### PR DESCRIPTION
**Pull request by mdh1418**
This is a follow up to PR #34662 
This PR consolidates the usage of hardcoded `Microsoft.NETCore.App` properties in `src/installer` and `src/libraries` with one common single shared property. It also removes unused properties with hardcoded `Microsoft.NETCore.App` and replaces occurences in other locations.
#35023

### CODE REVIEW

File: src/installer/pkg/projects/netcoreapp/sfx/Microsoft.NETCore.App.SharedFx.sfxproj [hunk 1]
```diff
@@ -2,8 +2,7 @@
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
 
   <PropertyGroup>
-    <SfxIdentity>Microsoft.NETCore.App</SfxIdentity>
```
Comment (dagood) [inline src/installer/pkg/projects/netcoreapp/sfx/Microsoft.NETCore.App.SharedFx.sfxproj:5]: Does this break usage (https://github.com/dotnet/arcade/blob/866b2acd5ffc3c2031c102f6415fd7c6a1a370d5/src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.sharedfx.targets#L97) or is this handled somewhere I'm not spotting?

File: eng/Versions.props [hunk 2]
```diff
@@ -139,7 +139,7 @@
     <LinuxCoreFxOptimizationDataPackage>optimization.linux-x64.ibc.corefx</LinuxCoreFxOptimizationDataPackage>
     <MicrosoftPrivateIntellisensePackage>microsoft.private.intellisense</MicrosoftPrivateIntellisensePackage>
     <MicrosoftDotNetXUnitConsoleRunnerPackage>Microsoft.DotNet.XUnitConsoleRunner</MicrosoftDotNetXUnitConsoleRunnerPackage>
-    <MicrosoftNETCoreAppPackage>Microsoft.NETCore.App</MicrosoftNETCoreAppPackage>
+    <MicrosoftNETCoreAppPackage>$(SharedFrameworkName)</MicrosoftNETCoreAppPackage>
```
Comment (dagood) [inline eng/Versions.props:142]: These props are truly for the package names, so I don't think this is necessary. Maestro++ might even revert this automatically, but I don't remember for sure.

(This corresponds to the package ID for `<MicrosoftNETCoreAppVersion>5.0.0-preview.4.20202.18</MicrosoftNETCoreAppVersion>`.)

File: src/installer/pkg/projects/netcoreapp/src/netcoreapp.depproj [hunk 3]
```diff
@@ -1,7 +1,6 @@
 <Project>
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
   <PropertyGroup>
-    <FrameworkPackageName>Microsoft.NETCore.App</FrameworkPackageName>
```
Comment (dagood) [inline src/installer/pkg/projects/netcoreapp/src/netcoreapp.depproj:4]: Where does this prop value come from now? (Note `SharedFrameworkName` vs. `FrameworkPackageName` and usages in [framework.dependency.targets](https://github.com/dotnet/arcade/blob/5966c5dd9deb5aca8fd8183eec454cf6cf864d14/src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.dependency.targets))

File: eng/Versions.props (hunk 2 above)
Comment (mdh1418) [reply eng/Versions.props]: Ah ok, reverted this change

File: src/installer/pkg/projects/netcoreapp/sfx/Microsoft.NETCore.App.SharedFx.sfxproj (hunk 1 above)
Comment (mdh1418) [reply src/installer/pkg/projects/netcoreapp/sfx/Microsoft.NETCore.App.SharedFx.sfxproj]: Ah I didn't realize it was being used in arcade as well. Reverting this change.

File: src/installer/pkg/projects/netcoreapp/src/netcoreapp.depproj (hunk 3 above)
Comment (mdh1418) [reply src/installer/pkg/projects/netcoreapp/src/netcoreapp.depproj]: I didn't see that it was being used in arcade, only grepped in dotnet/runtime. Reverting change

**Comment by mdh1418** [comment]
I don't believe the failing lanes are related to the changes. 
`Libraries Test Run release mono Windows_NT x64 Debug failed` had also been failing in #34662 
Could I get another look over the PR @safern @dagood ?

**Review by dagood — APPROVED** [review]

**Pull request merged**